src/app/download/download.service.ts

```diff
@@ -9,6 +9,16 @@ export class DownloadService {
     private readonly adAuthoringState: AdAuthoringWorkflowStateContainer
   ) {}
 
+  getDisabledVal() {
+    const file = this.adAuthoringState.getValue().file;
+    const landingUrl = this.adAuthoringState.getValue().landingUrl;
```

Comment:
You can use object destructuring here to prevent the repeated call to getValue()

```suggestion
    const {file, landingUrl} = this.adAuthoringState.getValue();
```

https://www.typescriptlang.org/docs/handbook/variable-declarations.html#object-destructuring